### PR TITLE
CNN Implementation

### PR DESCRIPTION
**Pull request by rkj26**
What has been done:
--- Implementation of the paper

Details of the model:
--- Used 10 parquet files for the model
--- Model trained for 50 epochs and a batch size of 16
--- Input Shape (100,200) that is 200 features and a timestep of 100.

Required Fixes/Improvements:
--- Improve memory usage so that all the data can be used.
--- Figure out why testing loss is not decreasing. Note: Training loss is converging.
--- Model is predicting 1 {Classes in the model is 0- decrease, 1 - stable,  2-increase} most of the time
--- Feature Engineering to improve the results.

### CODE REVIEW

**Comment by samholt** [comment]
Good start

The initial paper implementation looks like a good start

1. The normalisation seems incorrect, in the paper they normalise with the previous days standard deviation and mean for the current day, I believe this normalises using the whole dataset
2. Yes the model is clearly over-fitting, to avoid this you could try add regularisation in later adaptions of this implementation, or do some hyper parameter optimisation with k-fold cross validation on the dataset
2. a. I think the lowest hanging fruit to improve performance is tuning the alpha parameter to make sure that there is a sensible amount of examples of the targets (i.e. roughly equal examples of each class 0,1,2)
(As what I suspect at the moment is that the current alpha parameter is making a disproportionate class distribution of 1, and the model is mainly learning to predict 1, hence this explains why it mainly outputs 1)
